Google.Api.Gax.Grpc/ApiMetadata.cs

```diff
@@ -39,10 +40,17 @@ public sealed partial class ApiMetadata
         /// </summary>
         public string Name { get; }
 
-        private ApiMetadata(string name)
+        /// <summary>
+        /// When true, <see cref="RestGrpcAdapter"/> will request that enums are encoded as numbers in JSON
+        /// rather than as strings, preserving unknown values.
+        /// </summary>
+        public bool RequestNumericEnumJsonEncoding { get; }
+
+        private ApiMetadata(string name, Lazy<IReadOnlyList<FileDescriptor>> fileDescriptorsProvider, bool requestNumericEnumJsonEncoding)
```

Comment:
Or make this one recieve the `Func<IEnumerable<FileDescriptor>>` and build the lazy here.
You then only need to convert the `IEnumerable<FileDescriptor> descriptors` in `() => descriptors` before calling this constructor.

Reply:
That would change the laziness for that case though.
See https://github.com/googleapis/gax-dotnet/blob/main/Google.Api.Gax.Grpc.Tests/ApiMetadataTest.cs#L16 for tests that would fail - which admittedly look like they're named the wrong way round...

Reply:
Ah yes, I though that the non-laziness there was unintended.